### PR TITLE
#1915 tidy up indicator utilities

### LINKED ISSUE #1915: Tidy up indicator utilities
## Is your feature request related to a problem? Please describe.

Placeholder issue for tidying up indicator utilities.

## Describe the solution you'd like

General feature tidy up of unused variables, missing comments etc.

## Implementation

N/A.

## Describe alternatives you've considered

N/A.

## Additional context

N/A.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #1915. Branches off #1901.

## Change summary

Just a tidy up of indicators utilities for readability.

## Testing

Functional changes only.

- [ ] App does not crash.

### Related areas to think about

N/A.

### CODE REVIEW

File: src/pages/dataTableUtilities/getIndicatorTableData.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable prettier/prettier */
```
Comment (joshxg) [inline src/pages/dataTableUtilities/getIndicatorTableData.js]: whats that about

File: src/pages/dataTableUtilities/getIndicatorTableData.js [hunk 2]
```diff
@@ -36,6 +37,37 @@ const COLUMNS = {
...
+  const result = values.findIndex(findByValue(value)) > 0;
+  return result;
+};
+
+/**
+ * Returns a filter method for getting the intersection of two arrays of IndicatorValue objects.
+ * @param {Array.<IndicatorValue>} values
+ * @return {Func}
+ */
+const filterByValues = values => value => includesValue(values, value);
```
Comment (joshxg) [inline src/pages/dataTableUtilities/getIndicatorTableData.js]: could also use https://www.npmjs.com/package/set-manipulator (which we have installed)

File: src/pages/dataTableUtilities/getIndicatorTableData.js [hunk 3]
```diff
@@ -44,81 +76,90 @@ const COLUMNS = {
  * @param {Period} period
  */
 const initialiseRowColumnValue = (row, column, period) => {
-  UIDatabase.write(() => {
-    createRecord(UIDatabase, 'IndicatorValue', row, column, period);
-  });
+  UIDatabase.write(() => createRecord(UIDatabase, 'IndicatorValue', row, column, period));
+  return UIDatabase.objects('IndicatorValue')
+    .filtered('row == $0 AND column == $1 AND period == $2', row, column, period);
```
Comment (joshxg) [inline src/pages/dataTableUtilities/getIndicatorTableData.js]:
```
let indicator;
UIDatabase.write(() => { indicator = createRecord(UIDatabase, 'IndicatorValue', row, column, period) });
return indicator;
```

ugly but 🤷‍♂

Comment (wlthomson) [reply src/pages/dataTableUtilities/getIndicatorTableData.js]: Feels wrong but is a lot better performance-wise. 425a8d2d.

File: src/pages/dataTableUtilities/getIndicatorTableData.js (hunk 2 above)
Comment (wlthomson) [reply src/pages/dataTableUtilities/getIndicatorTableData.js]: I spent ages trying to do it with js, I didn't know we had this! Anyway, something for a later fix (with the selectors etc.).

File: src/pages/dataTableUtilities/getIndicatorTableData.js (hunk 1 above)
Comment (wlthomson) [reply src/pages/dataTableUtilities/getIndicatorTableData.js]: 7abbb235.

cookie

**Pull request merged**